aws/default_retryer.go

```diff
@@ -111,26 +109,10 @@ func (d DefaultRetryer) ShouldRetry(r *Request) bool {
 		return *r.Retryable
 	}
 
-	if r.HTTPResponse.StatusCode >= 500 {
-		return true
-	}
-	return r.IsErrorRetryable() || d.shouldThrottle(r)
-}
-
-// ShouldThrottle returns true if the request should be throttled.
-func (d DefaultRetryer) shouldThrottle(r *Request) bool {
-	if r.HTTPResponse != nil {
-		switch r.HTTPResponse.StatusCode {
-		case 429:
-		case 502:
-		case 503:
-		case 504:
-		default:
-			return r.IsErrorThrottle()
-		}
+	if r.HTTPResponse.StatusCode >= 500 && r.HTTPResponse.StatusCode != 501 {
```

Comment:
The status code determination for retry probably should happen in `r.IsErrorThrottle` since the `Request.IsErrorThrottle` method already checks status codes. would be better not to duplicate this here and `r.IsErrorThrottle`

Reply:
Retryable is true for all the status code >= 500 except 501. while r.isErrorThrottle is true only when status code is either 429, 502, 503,or 504.  Is this the correct behavior?

Reply:
Good point, I was incorrectly thinking `500` should be included in throttle. The `500` statuscode check should be a part of `r.IsErrorRetryable`  A 500 should be a regular retry, not throttle.